contracts/AxelarGateway.sol

```diff
@@ -551,7 +545,8 @@ contract AxelarGateway is IAxelarGateway, AdminMultisigBase {
     }
 
     function _getTokenDailyMintAmountKey(string memory symbol, uint256 day) internal pure returns (bytes32) {
-        return keccak256(abi.encodePacked(PREFIX_TOKEN_DAILY_MINT_AMOUNT, symbol, day));
+        // abi.encode to securely hash dynamic-length symbol data followed by day
+        return keccak256(abi.encode(PREFIX_TOKEN_DAILY_MINT_AMOUNT, symbol, day));
```

Comment:
Nice, alternatively, we encode the fixed length day first followed by symbol with `encodePacked` instead

Reply:
Since there is only one unfixed-length variable I don't think it matters actually: The old encodePacked was always 32b followed by an unknown length, followed by another 32b, Which is unique. If you have more than one multiple length variables however then thery can be manipulated to create the same encoding from different inputs.

Reply:
Ah, yes, although `encode` is just safer to reason about, so I'm ok with switching to it